text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -209,11 +291,53 @@ class Sampler {
         return out_token;
     }
 
+    int64_t _multinomial_sample(ov::Tensor logits, float temperature, float top_p, size_t top_k) {
+        ov::Shape logits_shape = logits.get_shape();
+        size_t batch_size = logits_shape[0], seq_len = logits_shape[1], vocab_size = logits_shape[2];
+        OPENVINO_ASSERT(batch_size == 1);
+
+        const float * logits_data = logits.data<const float>() + (seq_len - 1) * vocab_size;
+        std::vector<LogitWithIdx> logit_vector(vocab_size);
+        for (size_t i = 0; i < logit_vector.size(); i++) {
+            logit_vector[i] = LogitWithIdx(logits_data[i], i);
+        }
+
+        auto temperature_transform = TemperatureLogitTransform(temperature);
+        std::vector<ProbabilityWithIdx> softmax_vector = temperature_transform.apply(logit_vector);
+
+        std::vector<ProbabilityWithIdx> filtered(softmax_vector);
+
+        if (top_p != 0.0f) {
+            auto filter = TopPFilter(top_p);
+            filtered = filter.filter(filtered);
+        }
+
+        if (top_k != 0) {
+            auto filter = TopKFilter(top_k);
+            filtered = filter.filter(filtered);
+        }
```

Comment:
Are these parameters (`temperature`, `top_p`, `top_k`) applicable to multinomial sampling only?

Reply:
In the "beam search with multinomial sampling" approach, I suppose that these parameters can be reused with a similar effect. @ilya-lavrenov do we need to enable beam search w/ multinomial right now?

Reply:
> right now?

no, we can do it later